src/main/java/nextstep/subway/path/infra/JgraphtPathFinder.java

```diff
@@ -0,0 +1,58 @@
+package nextstep.subway.path.infra;
+
+import lombok.RequiredArgsConstructor;
+import nextstep.subway.path.domain.Path;
+import nextstep.subway.path.domain.PathFinder;
+import nextstep.subway.path.domain.PathSections;
+import nextstep.subway.path.domain.PathStation;
+import org.jgrapht.GraphPath;
+import org.jgrapht.alg.shortestpath.DijkstraShortestPath;
+import org.jgrapht.graph.DefaultWeightedEdge;
+import org.jgrapht.graph.WeightedMultigraph;
+import org.springframework.stereotype.Component;
+
+import java.util.Objects;
+
+@RequiredArgsConstructor
+@Component
+public class JgraphtPathFinder implements PathFinder {
+
+    @Override
+    public Path findShortest(final PathSections pathSections, final PathStation source, final PathStation target) {
+        validate(source, target);
+
+        WeightedMultigraph<PathStation, DefaultWeightedEdge> graph = new WeightedMultigraph<>(DefaultWeightedEdge.class);
+        DijkstraShortestPath<PathStation, DefaultWeightedEdge> dijkstraShortestPath = initGraph(graph, pathSections);
+        GraphPath<PathStation, DefaultWeightedEdge> shortestPath = dijkstraShortestPath.getPath(source, target);
+        if (isDisconnectedPath(shortestPath)) {
+            throw new IllegalArgumentException("연결되지 않은 지하철역 입니다.");
+        }
+
+        return new Path(shortestPath.getVertexList(), (int) shortestPath.getWeight());
+    }
+
+    private void validate(final PathStation source, final PathStation target) {
+        if (source.equals(target)) {
+            throw new IllegalArgumentException("출발역과 도착역이 동일합니다.");
+        }
+    }
+
+    private boolean isDisconnectedPath(final GraphPath<PathStation, DefaultWeightedEdge> path) {
+        return Objects.isNull(path);
+    }
+
+    private DijkstraShortestPath<PathStation, DefaultWeightedEdge> initGraph(
+            final WeightedMultigraph<PathStation, DefaultWeightedEdge> graph,
+            final PathSections pathSections) {
+        pathSections.getPathStations()
```

Comment:
```suggestion
        pathSections.forEach(graph::addVertex);
```

Consumer를 활용하면 getter로 값을 꺼내지 않고도 vertex 추가가 가능할거에요

Reply:
PathSections가 일급컬렉션 객체라 Consumer 사용에 제약이 있었습니다..! 

PathSections 내부에 로직을 구현함으로  getter를 사용하지 않는 방법도 존재할 것 같은데
메인 로직이 외부 라이브러리에 의존성을 갖게 되는 것 같아 지양하고 싶습니다. 🤔 

클래스의 구조를 변경하여 다시 생각해보아야 할까요..?
제가 현구님 피드백을 잘 이해하지 못한 것일 수도 있을 것 같습니다 😢

Reply:
외부 라이브러리 의존성을 낮추기 위해 PathFinder 인터페이스를 사용하고 Jgraph 구현체를 작성하신거 아닌가요?
최단거리 구하는 라이브러리를 교체하게 되면 PathFinder를 구현하는 다른 클래스가 추가되겠죠

Reply:
```java
    public void forEach(Consumer<PathStation> consumer) {
        getPathStations().forEach(consumer);
    }
```

이렇게 작성하면 PathSections 클래스에서 Jgraph를 의존할 일도 없고 getPathStations를 private으로 감출 수도 있겠네요

Reply:
일급 컬렉션에 Consumer를 사용한 forEach를 정의해주면 getter의 사용이 확 줄겠군요 ! 👍 👍 👍 👍 

피드백 감사드립니다 ! 🙇